Tangem/Modules/Markets/TokenMarketsDetails/Subviews/MarketsPortfolioContainerView/MarketsPortfolioTokenItemView.swift

```diff
@@ -32,31 +51,131 @@ struct MarketsPortfolioTokenItemView: View {
+                HStack(spacing: .zero) {
+                    Text(viewModel.walletName)
+                        .lineLimit(1)
+                        .style(Fonts.Bold.subheadline, color: Colors.Text.primary1)
+                }
+                .frame(minWidth: 0.3 * textBlockSize.width, alignment: .leading)
 
-                Spacer(minLength: 8)
+                Spacer(minLength: Constants.spacerLength)
 
                 Text(viewModel.fiatBalanceValue)
                     .lineLimit(1)
                     .style(Fonts.Regular.subheadline, color: Colors.Text.primary1)
             }
 
             HStack {
-                Text(viewModel.tokenName)
-                    .lineLimit(1)
-                    .style(Fonts.Regular.caption1, color: Colors.Text.tertiary)
+                HStack(spacing: .zero) {
+                    Text(viewModel.tokenName)
+                        .lineLimit(1)
+                        .style(Fonts.Regular.caption1, color: Colors.Text.tertiary)
+                }
+                .frame(minWidth: 0.32 * textBlockSize.width, alignment: .leading)
 
-                Spacer(minLength: 8)
+                Spacer(minLength: Constants.spacerLength)
 
                 Text(viewModel.balanceValue)
                     .truncationMode(.middle)
                     .lineLimit(1)
                     .style(Fonts.Regular.caption1, color: Colors.Text.tertiary)
             }
         }
+        .readGeometry(\.size, bindTo: $textBlockSize)
+    }
+
+    @ViewBuilder
+    private var background: some View {
+        if #available(iOS 16.0, *), let roundedCornersConfiguration = roundedCornersConfiguration {
+            Colors.Background.action
+                .cornerRadiusContinuous(
+                    topLeadingRadius: roundedCornersConfiguration.topLeadingRadius,
+                    bottomLeadingRadius: roundedCornersConfiguration.bottomLeadingRadius,
+                    bottomTrailingRadius: roundedCornersConfiguration.bottomTrailingRadius,
+                    topTrailingRadius: roundedCornersConfiguration.topTrailingRadius
+                )
+        } else {
+            Colors.Background.action
+        }
+    }
+
+    @ViewBuilder
+    private func contextMenuButton(for actionType: TokenActionType) -> some View {
+        let action = { viewModel.didTapContextAction(actionType) }
+        if actionType.isDestructive {
+            Button(
+                role: .destructive,
+                action: action,
+                label: {
+                    labelForContextButton(with: actionType)
+                }
+            )
+        } else {
+            Button(action: action, label: {
+                labelForContextButton(with: actionType)
+            })
+        }
+    }
+
+    private func labelForContextButton(with action: TokenActionType) -> some View {
+        HStack {
+            Text(action.title)
+            action.icon.image
+                .renderingMode(.template)
```

Comment:
слиплись немного...
```suggestion
            Text(action.title)
            
            action.icon.image
                .renderingMode(.template)
```

я не помню, а `renderingMode` добавляешь чтобы за счет роли `destructive` покрасилась иконка, верно? а то там вроде нигде цвет не задается.

Reply:
Насколько знаю да, реализация с TokenItemView